packages/back-end/src/models/FeatureModel.ts

```diff
@@ -367,6 +367,24 @@ export async function deleteAllFeaturesForAProject({
   }
 }
 
+/**
+ * Deletes rules for a given environment from all features
+ */
+export async function removeEnvironmentFromFeatureRules(
+  context: ReqContext | ApiReqContext,
+  envId: string
+) {
+  const environmentKey = `environmentSettings.${envId}`;
+  const query = {
+    organization: context.org.id,
+    [environmentKey]: { $exists: true },
+  };
+
+  await FeatureModel.updateMany(query, {
+    $unset: { [environmentKey]: "" },
+  });
```

Comment:
@jdorn are we good with this sort of raw deletion? It doesn't leave an audit log or create a revision for the feature, but I was thinking it makes more sense to do it this way because it's a cascading delete from the environment itself. I'm also worried about performance if we do all that overhead work, but I figured I should run that past you

Reply:
Chatted w/ Jeremy about this - we're good with this being a direct delete (no audit, no revision) as long as it's an option (default false) for users to opt into.